src/pageComponents/home.js

```diff
@@ -0,0 +1,111 @@
+import { topRatedMovieUrl, topRatedTVShowsUrl, genreUrl } from "../movies/constants/Constants";
+import React, { useEffect, useState } from "react";
+import TopRatedMovie from "../movies/api/GetTopRated";
+import ActionsButtons from "../genericComponents/Carousel";
+
+export default function Home() {
+  const [respTopRatedMovie, setRespTopRatedMovie] = useState([]);
+  const [respTopRatedTVShows, setResprespTopRatedTVShows] = useState([]);
+
+  //top rated movie
+  useEffect(() => {
+    const resultTopRatedMovie = TopRatedMovie(topRatedMovieUrl);
+    const resultMovieGenre = TopRatedMovie(genreUrl);
+    DataSettingForCarouselFilmDetail(
+      {
+        resultTopRated: resultTopRatedMovie,
+        resultMovieGenre: resultMovieGenre,
+      },
+      setRespTopRatedMovie
+    );
+  }, []);
+
+  //top rated tv series
+  useEffect(() => {
+    const resultTopRatedTVS = TopRatedMovie(topRatedTVShowsUrl);
+    const resultMovieGenre = TopRatedMovie(genreUrl);
+    DataSettingForCarouselFilmDetail({ resultTopRated: resultTopRatedTVS, resultMovieGenre: resultMovieGenre }, setResprespTopRatedTVShows);
+  }, []);
+
+  return (
+    <main>
+      <h1>Top Rated Movie</h1>
+      <hr></hr>
+      <ActionsButtons data={respTopRatedMovie} />
+
+      <h1>Top Rated TV Series</h1>
+      <hr></hr>
+      <ActionsButtons data={respTopRatedTVShows} />
+    </main>
+  );
+}
+
+//this function prepares data for the film detail after the film selection on carousel
+export function DataSettingForCarouselFilmDetail(data, setState) {
+  data.resultTopRated
+    .then((TRTVSresponse) => {
+      data.resultMovieGenre.then((genResponse) => {
+        const newDataWithGenreAdd = JoinGenreinData({
+          genre: genResponse,
+          topRated: TRTVSresponse,
+        });
+        const dataReadyTVSForGenres = DataSettingForCarousel(newDataWithGenreAdd);
+        setState(dataReadyTVSForGenres);
+      });
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+}
+
+//add index for carousel (see ActionsButtons in GenericComponent for more details)
+export function DataSettingForCarousel(props) {
+  let arrayModifiedForCarousel = props.map((elem) => {
+    return {
+      ...elem,
+      index: props.indexOf(elem),
+    };
+  });
+  return arrayModifiedForCarousel;
```

Comment:
Essendo una funzione il nome inizia sempre con la minuscola, e descrivere in modo significativo il parametro della funziona props in questo caso è errato.